pkg/gui/diff_context_size_test.go

```diff
@@ -0,0 +1,166 @@
+		gui.pushContextDirect(context)
+
+		gui.IncreaseContextInDiffView()
+
+		assert.Equal(t, 1, gui.Config.GetUserConfig().Git.DiffContextSize, string(context.GetKey()))
+	}
+}
+
+func TestDecreasesContextInDiffViewByOneInContextWithDiff(t *testing.T) {
+	contexts := []func(gui *Gui) Context{
+		func(gui *Gui) Context { return gui.State.Contexts.Files },
+		func(gui *Gui) Context { return gui.State.Contexts.BranchCommits },
+		func(gui *Gui) Context { return gui.State.Contexts.CommitFiles },
+		func(gui *Gui) Context { return gui.State.Contexts.Stash },
+		func(gui *Gui) Context { return gui.State.Contexts.Staging },
+		func(gui *Gui) Context { return gui.State.Contexts.PatchBuilding },
+		func(gui *Gui) Context { return gui.State.Contexts.SubCommits },
+	}
+
+	for _, c := range contexts {
+		gui := NewDummyGui()
+		context := c(gui)
+		setupGuiForTest(gui)
+		gui.Config.GetUserConfig().Git.DiffContextSize = 2
+		gui.pushContextDirect(context)
+
+		gui.DecreaseContextInDiffView()
+
+		assert.Equal(t, 1, gui.Config.GetUserConfig().Git.DiffContextSize, string(context.GetKey()))
+	}
+}
+
+func TestDoesntDecreaseContextInDiffViewInContextWithoutDiff(t *testing.T) {
+	contexts := []func(gui *Gui) Context{
+		func(gui *Gui) Context { return gui.State.Contexts.Status },
+		func(gui *Gui) Context { return gui.State.Contexts.Submodules },
+		func(gui *Gui) Context { return gui.State.Contexts.Remotes },
+		func(gui *Gui) Context { return gui.State.Contexts.Normal },
+		func(gui *Gui) Context { return gui.State.Contexts.ReflogCommits },
+		func(gui *Gui) Context { return gui.State.Contexts.RemoteBranches },
+		func(gui *Gui) Context { return gui.State.Contexts.Tags },
+		func(gui *Gui) Context { return gui.State.Contexts.Merging },
+		func(gui *Gui) Context { return gui.State.Contexts.CommandLog },
+	}
+
+	for _, c := range contexts {
+		gui := NewDummyGui()
+		context := c(gui)
+		setupGuiForTest(gui)
+		gui.Config.GetUserConfig().Git.DiffContextSize = 2
+		gui.pushContextDirect(context)
+
+		gui.DecreaseContextInDiffView()
+
+		assert.Equal(t, 2, gui.Config.GetUserConfig().Git.DiffContextSize, string(context.GetKey()))
+	}
+}
+
+func TestDoesntIncreaseContextInDiffViewInContextWhenInPatchBuildingMode(t *testing.T) {
+	gui := NewDummyGui()
+	setupGuiForTest(gui)
+	gui.Config.GetUserConfig().Git.DiffContextSize = 2
+	gui.pushContextDirect(gui.State.Contexts.CommitFiles)
+	gui.GitCommand.PatchManager.Start("from", "to", false, false)
+
+	gui.IncreaseContextInDiffView()
+
+	assert.Equal(t, 2, gui.Config.GetUserConfig().Git.DiffContextSize)
+}
+
+func TestDoesntDecreaseContextInDiffViewInContextWhenInPatchBuildingMode(t *testing.T) {
+	gui := NewDummyGui()
+	setupGuiForTest(gui)
+	gui.Config.GetUserConfig().Git.DiffContextSize = 2
+	gui.pushContextDirect(gui.State.Contexts.CommitFiles)
+	gui.GitCommand.PatchManager.Start("from", "to", false, false)
+
+	gui.DecreaseContextInDiffView()
+
+	assert.Equal(t, 2, gui.Config.GetUserConfig().Git.DiffContextSize)
```

Comment:
it would be better here for me to actually assert that an error was raised but I'll leave that for a future PR

Reply:
Actually turns out this needs to be dealt with now because CI is panicking due to the attempted error panel popup. I've introduced a PopupHandler interface which we swap out in the test